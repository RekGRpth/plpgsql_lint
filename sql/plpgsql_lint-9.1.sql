/home/pavel/src/postgresql-9.0.5/contrib/plpgsql_lint/sql/plpgsql_lint-9.0.sql