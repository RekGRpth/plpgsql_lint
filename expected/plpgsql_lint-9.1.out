load 'plpgsql';
load 'plpgsql_lint';
create table t1(a int, b int);
create function f1()
returns void as $$
begin
  if false then
    update t1 set c = 30;
  end if;
end;
$$ language plpgsql;
select f1();
ERROR:  column "c" of relation "t1" does not exist
LINE 1: update t1 set c = 30
                      ^
QUERY:  update t1 set c = 30
CONTEXT:  PL/pgSQL function "f1" line 4 at SQL statement
drop function f1();
create function g1(out a int, out b int)
as $$
  select 10,20;
$$ language sql;
create function f1()
returns void as $$
declare r record;
begin
  r := g1();
  if false then 
    raise notice '%', r.c;
  end if;
end;
$$ language plpgsql;
select f1();
ERROR:  record "r" has no field "c"
CONTEXT:  SQL statement "SELECT r.c"
PL/pgSQL function "f1" line 6 at RAISE
drop function f1();
drop function g1();
create function g1(out a int, out b int)
returns setof record as $$
select * from t1;
$$ language sql;
create function f1()
returns void as $$
declare r record;
begin
  for r in select * from g1()
  loop
    raise notice '%', r.c;
  end loop;
end;
$$ language plpgsql;
select f1();
ERROR:  record "r" has no field "c"
CONTEXT:  SQL statement "SELECT r.c"
PL/pgSQL function "f1" line 6 at RAISE
create or replace function f1()
returns void as $$
declare r record;
begin
  for r in select * from g1()
  loop
    r.c := 20;
  end loop;
end;
$$ language plpgsql;
select f1();
ERROR:  record "r" has no field "c"
CONTEXT:  PL/pgSQL function "f1" line 4 at FOR over SELECT rows
drop function f1();
drop function g1();
create function f1()
returns int as $$
declare r int;
begin
  if false then
    r := a + b;
  end if;
  return r;
end;
$$ language plpgsql;
select f1();
ERROR:  column "a" does not exist
LINE 1: SELECT a + b
               ^
QUERY:  SELECT a + b
CONTEXT:  PL/pgSQL function "f1" line 5 at assignment
drop function f1();
create or replace function f1()
returns void as $$
begin
  if false then
    raise notice '%', 1, 2;
  end if;
end;
$$ language plpgsql;
select f1();
ERROR:  too many parameters specified for RAISE
CONTEXT:  PL/pgSQL function "f1" line 4 at RAISE
drop function f1();
create or replace function f1()
returns void as $$
begin
  if false then
    raise notice '% %';
  end if;
end;
$$ language plpgsql;
select f1();
ERROR:  too few parameters specified for RAISE
CONTEXT:  PL/pgSQL function "f1" line 4 at RAISE
drop function f1();
create or replace function f1()
returns void as $$
declare r int[];
begin
  if false then
    r[c+10] := 20;
  end if;
end;
$$ language plpgsql;
select f1();
ERROR:  column "c" does not exist
LINE 1: SELECT c+10
               ^
QUERY:  SELECT c+10
CONTEXT:  PL/pgSQL function "f1" line 5 at assignment
drop function f1();
create or replace function f1()
returns void as $$
declare r int;
begin
  if false then
    r[10] := 20;
  end if;
end;
$$ language plpgsql;
select f1();
ERROR:  subscripted object is not an array
CONTEXT:  PL/pgSQL function "f1" line 5 at assignment
drop function f1();
create type diagnostic_info_type as (
  status text,
  message text,
  detail text,
  row_count int);
create or replace function f1()
returns void as $$
declare
  dg record;
begin
  dg := NULL::diagnostic_info_type;
  if false then
    dg.status := '00000';
    dg.mistake := 'hello';
  end if;
end;
$$ language plpgsql;
select f1();
ERROR:  record "dg" has no field "mistake"
CONTEXT:  PL/pgSQL function "f1" line 7 at assignment
drop function f1();
